lib/iris/_merge.py

```diff
@@ -1263,9 +1285,15 @@ def register(self, cube, error_on_mismatch=False):
             this :class:`ProtoCube`.
 
         """
-        match = self._cube_signature.match(self._build_signature(cube),
-                                           error_on_mismatch)
+        signature = self._cube_signature
+        other = self._build_signature(cube)
+        match = signature.match(other, error_on_mismatch)
         if match:
+            if signature.fill_value is None and other.fill_value is not None:
```

Comment:
No. There is no need in that case. We only want promotion to occur when the proto-cube `fill_value` is `None` and the other candidate cube `fill_value` is not `None` i.e. that's the case when we want to promote the `fill_value` from the candidate cube to the proto-cube.